javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -218,7 +219,7 @@ public final Node setComment(final Comment comment) {
     }
 
     public boolean hasJavaDocComment() {
-        return hasComment() && getComment() instanceof JavadocComment;
+        return hasComment() && getComment().get() instanceof JavadocComment;
```

Comment:
@arturbosch if you have experience with metamodels in general, you could take a look at what I wrote above to see if it makes sense.